ibmcloud/image/build.sh

```diff
@@ -206,69 +257,163 @@ umount "$src_mnt/dev"
 umount "$src_mnt/proc"
 umount "$src_mnt/sys"
 
-echo -e "\nCopying the root filesystem\n"
-
-label=$(lsblk -n -o label "$src_part")
-mkfs.ext4 -L "$label" "$dst_part"
 mkdir -p "$dst_mnt"
-mount "$dst_part" "$dst_mnt"
+if [ "${SE_BOOT}" = "1" ]; then
+    dst_part="${tmp_nbd}p2"
+    label="root"
+    printf "\nFormatting root partition\n"
+    mkfs.ext4 -L "$label" /dev/mapper/$LUKS_NAME
+    mount /dev/mapper/$LUKS_NAME "$dst_mnt"
+    mkdir ${dst_mnt}/etc
+    mkdir ${dst_mnt}/boot-se
+    mount -o norecovery ${tmp_nbd}p1 ${dst_mnt}/boot-se
+else
+    label=$(lsblk -n -o label "$src_part")
+    mkfs.ext4 -L "$label" "$dst_part"
+    mount "$dst_part" "$dst_mnt"
+fi
 
+echo "Copying the root filesystem"
 tar_opts=(--numeric-owner --preserve-permissions --acl --selinux --xattrs --xattrs-include='*' --sparse)
 tar -cf - "${tar_opts[@]}" --sort=none -C "$src_mnt" . | tar -xf - "${tar_opts[@]}" --preserve-order  -C "$dst_mnt"
 
+echo "The root filesystem is ready"
+sleep 1
 umount "$src_mnt"
-
 mount -t sysfs sysfs "$dst_mnt/sys"
 mount -t proc proc "$dst_mnt/proc"
 mount --bind /dev "$dst_mnt/dev"
 
-case "$(uname -m)" in
-    s390x)
-        echo -e "\nExecuting zipl\n"
-        helper="$dst_mnt/lib/s390-tools/zipl_helper.nbd"
-        cat <<END > "$helper"
+if [ "${SE_BOOT}" = "1" ]; then
```

Comment:
It's not necessary in this PR, but if we want to add similar function for libvirt, maybe need consider separate SE related function and place it somewhere can be reused like https://github.com/confidential-containers/cloud-api-adaptor/tree/9e976024cb9927d17e077f3de6a2a8e3de0a4725/podvm/qcow2? so that libvirt provider on LinuxONE can also leverage the function.